### PR TITLE
feat(neptune-alpha): specify port for the cluster

### LINKED ISSUE #31074: neptune: specify port for the cluster
### Describe the feature

Specify port number of the Neptune cluster.

### Use Case

We can configure the port number of Neptune cluster but CDK L2 construct cannot configure this.

### Proposed Solution

Add `port` property to `DatabaseClusterProps` and set DBPort to the `port` value.
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-neptune-dbcluster.html#cfn-neptune-dbcluster-dbport

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.150.0

### Environment details (OS name and version, etc.)

irrelevant

### PR DESCRIPTION
**Pull request by badmintoncryer**
### Issue # (if applicable)

Closes #31074.

### Reason for this change

Cloudformation supports for setting port number for the Neptune cluster but AWS CDK cannot do this.

### Description of changes

Add `port` prop to `DatabaseClusterProps`.

### Description of how you validated changes

Added both unit and integ tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-neptune-alpha/lib/cluster.ts [hunk 1]
```diff
@@ -332,6 +332,13 @@ export interface DatabaseClusterProps {
    * @default - false
    */
   readonly copyTagsToSnapshot?: boolean;
+
+  /**
+   * The port number
```
Comment (go-to-k) [inline packages/@aws-cdk/aws-neptune-alpha/lib/cluster.ts:337]: It is better to be more helpful.

```suggestion
   * The port number on which the DB instances in the DB cluster accept connections.
```

**Review by go-to-k — CHANGES_REQUESTED** [review]
No problems at all, just very detailed comments.

**Comment by badmintoncryer** [comment]
Thank you @go-to-k !
I've updated JSdoc as you suggested.

**Review by go-to-k — APPROVED** [review]
Thanks. LGTM.

File: packages/@aws-cdk/aws-neptune-alpha/lib/cluster.ts [hunk 2]
```diff
@@ -332,6 +332,13 @@ export interface DatabaseClusterProps {
    * @default - false
    */
   readonly copyTagsToSnapshot?: boolean;
+
+  /**
+   * The port number on which the DB instances in the DB cluster accept connections.
+   *
+   * @default 8182
+   */
+  readonly port? :number;
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-neptune-alpha/lib/cluster.ts:341]:
```suggestion
  readonly port?: number;
```

**Review by kaizencc — APPROVED** [review]

**Pull request merged**